### PR TITLE
feat(login): support client side only logout

### PR DESCRIPTION
**Pull request by efiShtain**
we add a non http only cookie named ak_logged_in
so frontend can remove it when user logs out
we always validate both cookies exists
in this way we keep the authenticated cookie secured and allow the frontend to logout without additional api on the backend side

### CODE REVIEW

File: internal/backend/auth/authhttpmiddleware/authmiddleware.go [hunk 1]
```diff
@@ -83,6 +83,11 @@ func newSessionsMiddleware(next http.Handler, sessions authsessions.Store) http.
 				return
 			}
 
+			if session == nil {
```
Comment (itayd) [inline internal/backend/auth/authhttpmiddleware/authmiddleware.go]: why is this needed? if we error out here it wont get to the tokens middleware.

Comment (efiShtain) [reply internal/backend/auth/authhttpmiddleware/authmiddleware.go]: we have a problematic case

both session and tokens are optional

```go
	Sessions authsessions.Store `optional:"true"`
	Tokens   authtokens.Tokens  `optional:"true"`
```

so if tokens are optional, sessions has to validate or fail if session is invalid

it also makes an order dependency which is not explicit

we should refactor both checks to one function and make it super explicit and readable, no need for optional checks

```go
if session == nil && token == nil {
   return 401
}
```

what do you want to do ? fix it on this PR ? merge it and open another PR ?

File: internal/backend/auth/authhttpmiddleware/authmiddleware.go [hunk 2]
```diff
@@ -83,6 +83,11 @@ func newSessionsMiddleware(next http.Handler, sessions authsessions.Store) http.
...
+				http.Error(w, "invalid session", http.StatusUnauthorized)
+				return
+			}
+
 			if session != nil {
```
Comment (tebeka) [inline internal/backend/auth/authhttpmiddleware/authmiddleware.go]: Due to the `if` above, you don't need this one.

**Review by tebeka — APPROVED** [review]
Minor comment, also missing tests.

File: internal/backend/auth/authhttpmiddleware/authmiddleware.go (hunk 1 above)
Comment (efiShtain) [reply internal/backend/auth/authhttpmiddleware/authmiddleware.go]: my bad, it should work, I'll recheck required config

Comment (efiShtain) [reply internal/backend/auth/authhttpmiddleware/authmiddleware.go]: seems legit on  Itay's debug dashboard as well

File: internal/backend/auth/authhttpmiddleware/authmiddleware.go (hunk 2 above)
Comment (efiShtain) [reply internal/backend/auth/authhttpmiddleware/authmiddleware.go]: removed this section from code

**Comment by efiShtain** [comment]
> Minor comment, also missing tests.

@tebeka 
I've added tests, not happy with them, but at least we have them
could you please review and give your thoughts ? 

not table tests since I've used lots of different mocks and it's hard with table testing
I'm thinking about refactoring stuff in the future to make stuff more clear and testable without so much mocks

probably injecting the internal session store instead of init it in the new function

**Review by tebeka — COMMENTED** [review]
LGTM
